src/lib/util/rbtree.c

```diff
@@ -1122,6 +1133,23 @@ void *rbtree_iter_next_postorder(fr_rb_tree_iter_t *iter)
 	return x->data;
 }
 
+/** Remove the current node from the tree
+ *
+ * @note Only makes sense for in-order traversals.
+ *
+ * @param[in] iter	previously initialised with #rbtree_iter_init
+ */
+void rbtree_iter_delete(fr_rb_tree_iter_t *iter)
+{
+	fr_rb_node_t *x = iter->node;
+
+	if (unlikely(x == NIL)) return;
+	(void) rbtree_iter_next_inorder(iter);
```

Comment:
Could you fixup the iterator initialisers to insert a copy of the iterator function pointer in the iterator struct and call that instead.  Then this code should work for all iterator types.

Reply:
Will do (with the analogous change to the next functions you mention in the other comment), and definitely try the same test for the pre and post with delete.